target/linux/mediatek/image/mt7622.mk

```diff
@@ -183,10 +185,24 @@ define Device/dlink_eagle-pro-ai-m32-a1
   IMAGES += tftp.bin recovery.bin
   IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
   IMAGE/tftp.bin := append-kernel | pad-to $$(KERNEL_SIZE) | append-ubi | check-size
-  IMAGE/recovery.bin := append-kernel | pad-to $$(KERNEL_SIZE) | append-ubi | pad-to $$(IMAGE_SIZE) | m32-recovery-header-kernel1
+endef
+
+define Device/dlink_eagle-pro-ai-m32-a1
+  $(Device/dlink_eagle-pro-ai-ax3200-a1)
+  DEVICE_MODEL := EAGLE PRO AI M32
+  DEVICE_DTS := mt7622-dlink-eagle-pro-ai-m32-a1
+  IMAGE/recovery.bin := append-kernel | pad-to $$(KERNEL_SIZE) | append-ubi | pad-to $$(IMAGE_SIZE) | m32-r32-recovery-header-kernel1 DLK6E6010001
 endef
 TARGET_DEVICES += dlink_eagle-pro-ai-m32-a1
 
+define Device/dlink_eagle-pro-ai-r32-a1
+  $(Device/dlink_eagle-pro-ai-ax3200-a1)
+  DEVICE_MODEL := EAGLE PRO AI R32
+  DEVICE_DTS := mt7622-dlink-eagle-pro-ai-r32-a1
+  IMAGE/recovery.bin := append-kernel | pad-to $$(KERNEL_SIZE) | append-ubi | pad-to $$(IMAGE_SIZE) | m32-r32-recovery-header-kernel1 DLK6E6015001
```

Comment:
No `tftp.bin` for the r32 like we got for the m32?

I kinda find the `recovery.bin` naming inconsistent with the existing use of that image name suffix (which hints that the file is used *as* recovery image rather than to be flashed *by* a vendor's recovery system), but that already happened now...

Reply:
The tftp image is created for both devices. It's defined in Device/dlink_eagle-pro-ai-ax3200-a1 and used by Device/dlink_eagle-pro-ai-m32-a1 and Device/dlink_eagle-pro-ai-r32-a1.
Any suggestions for the naming of the recovery image? I took over the naming from D-Link DAP-X1860.

Reply:
Wouldn't `initramfs.bin` be considered the recovery for OpenWrt?
AFAIK `recovery.bin` is for vendor uboot recovery in several devices already.